qiskit/tools/parser.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Helper function to parse string expression given by backends."""
+
+import ast
+import operator
+import re
+
+import cmath
+
+from qiskit.exceptions import QiskitError
+
+
+# valid functions
+_math_ops = {key: value for (key, value) in vars(cmath).items() if key[0] != '_'}
```

Comment:
I think this should be replaced with an explicit list of allowed functions from `cmath`.